Controls/Chart/Chart.UWP/Visualization/Common/ContainerVisualsFactory.cs

```diff
@@ -266,21 +266,41 @@ protected internal virtual ContainerVisual PrepareCartesianAxisLineVisual(Cartes
         protected internal virtual ContainerVisual PrepareLineRenderVisual(ContainerVisual containerVisual, IEnumerable<Point> points, Brush stroke, double strokeThickness)
         {
             containerVisual.Children.RemoveAll();
-            for (int i = 0; i < points.Count(); i++)
+
+            using (var enumerator = points.GetEnumerator())
             {
-                SpriteVisual childSpiteVisual = containerVisual.Compositor.CreateSpriteVisual();
-                var startPoint = points.ElementAt(i);
-                if (i + 1 < points.Count())
+                var haveStartPoint = false;
+                var startPoint = default(Point);
+
+                while (enumerator.MoveNext())
                 {
-                    var endPoint = points.ElementAt(i + 1);
+                    var endPoint = enumerator.Current;
+
+                    if (!haveStartPoint)
+                    {
+                        startPoint = endPoint;
+                        if (!enumerator.MoveNext()) break;
+                        haveStartPoint = true;
+                        endPoint = enumerator.Current;
+                    }
+
+                    // draw the line if we have both points
+                    var childSpiteVisual = containerVisual.Compositor.CreateSpriteVisual();
                     var angle = Math.Atan2(endPoint.Y - startPoint.Y, endPoint.X - startPoint.X) * (180.0 / Math.PI);
-                    var width = Math.Sqrt((Math.Pow(startPoint.X - endPoint.X, 2) + Math.Pow(startPoint.Y - endPoint.Y, 2)));
+                    var width = Math.Sqrt(Math.Pow(startPoint.X - endPoint.X, 2) + Math.Pow(startPoint.Y - endPoint.Y, 2));
 
-                    childSpiteVisual.RotationAngleInDegrees = (float)angle;
+	                if (Math.Abs(angle) > 40)
+		                width += 2;
+	                else
+		                width++;
+
```

Comment:
Ouch... That was just a quick and dirty attempt to fix the gaps issue, I forgot to delete those lines during PR preparations.